code/modules/media/jukebox.dm

```diff
@@ -648,6 +668,26 @@ var/global/list/loopModeNames=list(
 						return
 			update_music()
 
+/obj/machinery/media/jukebox/proc/eject(var/playlist_name)
+	if(playlists.Find(playlist_name))
+		new /obj/item/weapon/vinyl(get_turf(src), playlist_name, playlists[playlist_name])
```

Comment:
Why are we creating on every eject instead of pulling from internal storage or item pooling?

Reply:
We could do it that way, but I will say this

When I refactored tank storage units to work the way you describe (generating all of the internal objects at shift start and keeping them inside instead of creating on eject), PJB complained at me because he said there was no reason to create objects before they're needed.

Is this really something we need to pool? I'm not seeing these as objects as ubiquitous.

Reply:
Maybe i'm just behind the times then.